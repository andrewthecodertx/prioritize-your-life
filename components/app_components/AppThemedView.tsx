import { View, type ViewProps } from "react-native";
import { useThemeColor } from "@/hooks/useThemeColor";

export type ThemedViewProps = ViewProps & {
  darkColor?: string;
  lightColor?: string;
};

/**
 * AppThemedView component that renders a View with a background color based on the current theme.
 *
 * @param style - The style object to be applied to the View component.
 * @param lightColor - The background color to be used when the theme is set to light.
 * @param darkColor - The background color to be used when the theme is set to dark.
 * @param otherProps - Additional props to be spread onto the View component.
 * @returns The rendered ThemedView component.
 */
export function AppThemedView({
  style,
  lightColor,
  darkColor,
  ...otherProps
}: ThemedViewProps) {
  const backgroundColor = useThemeColor(
    { light: lightColor, dark: darkColor },
    "background"
  );

  return <View style={[{ backgroundColor }, style]} {...otherProps} />;
}
