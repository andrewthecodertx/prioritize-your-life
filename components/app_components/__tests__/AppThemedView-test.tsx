import * as React from "react";
import renderer from "react-test-renderer";
import { AppThemedView } from "../AppThemedView";

it(`renders correctly`, () => {
  const tree = renderer.create(<AppThemedView />).toJSON();

  expect(tree).toMatchSnapshot();
});