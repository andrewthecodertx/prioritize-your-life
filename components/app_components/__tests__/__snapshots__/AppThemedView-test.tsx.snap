// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "#000000",
      },
      undefined,
    ]
  }
/>
`;
